### PR TITLE
DNMY: MOI

### PR DESCRIPTION
**Pull request by odow**
This PR updates GLPK for the latest version of MOI, and copies the files from https://github.com/JuliaOpt/MathOptInterfaceGLPK.jl.

### Blockers
- [x] Tag a release prior to merging this
- [x] Decide what to do with LinQuadOptInterface
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/5
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/6
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/7
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/9
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/10
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/17
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/24
- [x] Requires https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/29
- [x] Requires https://github.com/JuliaOpt/MathOptInterface.jl/pull/329
- [x] Requires https://github.com/JuliaOpt/MathOptInterface.jl/pull/354
- [x] Update MOI tag in REQUIRE

### Questions

- [x] Should we move `GLPKMathProgInterface.jl` to this repo?

### CODE REVIEW

**Comment by mlubin** [comment]
@carlobaldassi, note this is reverting our decision from 2013 to split off the JuMP-related code into a separate package. Things have changed a bit since then :).
Our current preference is to host the MathOptInterface code into the solver-wrappers for ease of maintainability and testing, unless there's a good reason not to.

@odow I don't see a need to move the code from GLPKMathProgInterface into here. It's just extra churn for code that soon won't be relevant.

**Comment by odow** [comment]
@mlubin Do you want to tag a release before this PR gets merged (or give me commit access :) )?

**Comment by mlubin** [comment]
@odow, all yours :)

**Comment by rdeits** [comment]
It looks like there's an attribute missing from the GLPKOptimizerLP(). Using this branch (and JuMP and MOI master), I can set up a simple LP: 

```julia
m = Model(optimizer=GLPKOptimizerLP())
@variable m x >= 0
@objective m Min x
JuMP.optimize(m)
```

which fails with:

```
Failed to copy model into optimizer: Unsupported attribute MathOptInterface.ObjectiveFunction{MathOptInterface.SingleVariable}()

Stacktrace:
 [1] optimize!(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.Utilities.UniversalFallback{JuMP.JuMPMOIModel{Float64}}}) at /home/rdeits/locomotion/explorations/jump-lcp/packages/v0.6/MathOptInterface/src/Utilities/cachingoptimizer.jl:155
 [2] #optimize#95(::Bool, ::Function, ::JuMP.Model) at /home/rdeits/locomotion/explorations/jump-lcp/packages/v0.6/JuMP/src/optimizerinterface.jl:48
 [3] optimize(::JuMP.Model) at /home/rdeits/locomotion/explorations/jump-lcp/packages/v0.6/JuMP/src/optimizerinterface.jl:37
```

If I convert the objective from a single variable to a `Linear` with:

```julia
@objective m Min x + 1
```

then everything works.

File: test/MOIWrapper.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
+using GLPK, Base.Test, MathOptInterface, MathOptInterface.Test
+
+const MOIT = MathOptInterface.Test
+
+@testset "MathOptInterfaceGLPK" begin
+    @testset "Unit Tests" begin
+        @testset "LP solver" begin
+            config = MOIT.TestConfig()
+            solver = GLPKOptimizerLP()
+            MOIT.unittest(solver, config, ["solve_singlevariable_obj"])
```
Comment (odow) [inline test/MOIWrapper.jl]: @rdeits yes, single variable objectives aren't supported yet. They need to be implemented in LQOI.

There is a variety of other stuff not implemented yet as well. See
https://github.com/JuliaOpt/LinQuadOptInterface.jl/issues/8

**Comment by rdeits** [comment]
Got it, thanks!

**Comment by odow** [comment]
Hmm @mlubin @joaquimg we have a problem. Because the MOI tests incorrectly allow adding multiple integer constraints (see https://github.com/JuliaOpt/MathOptInterface.jl/issues/404, https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/30, https://github.com/JuliaOpt/LinQuadOptInterface.jl/pull/31), this PR now fails tests.

We can either 
- disable the failing tests here
- fix MOI tests and push a new release
- merge as is with failing tests.

Thoughts?

**Comment by mlubin** [comment]
Both "disable the failing tests here" (with a TODO explaining why) and "fix MOI tests and push a new release" are viable options.

**Comment by blegat** [comment]
We can target v0.7 for a next release

**Comment by odow** [comment]
This also broke 32-bit because of https://github.com/JuliaOpt/MathOptInterface.jl/issues/377

**Pull request merged**